### ISSUE TITLE
Patch the image uboot to filter noise on the bbb rx line

### ISSUE THREAD

**Issue by BrianAdams** [OpenROV/openrov-software#467]
Per https://github.com/OpenROV/openrov-electronics/issues/32

I'm not sure how uboot ends up on the image. There is a clear part of the chroot script that pulls the source code down, but I don't see any compile calls.
 

```
other_source_links () {
	rcn_https="https://rcn-ee.com/repos/git/u-boot-patches"

	mkdir -p /opt/source/u-boot_${u_boot_release}/
	wget --directory-prefix="/opt/source/u-boot_${u_boot_release}/" ${rcn_https}/${u_boot_release}/0001-omap3_beagle-uEnv.txt-bootz-n-fixes.patch
	wget --directory-prefix="/opt/source/u-boot_${u_boot_release}/" ${rcn_https}/${u_boot_release}/0001-am335x_evm-uEnv.txt-bootz-n-fixes.patch
	mkdir -p /opt/source/u-boot_${u_boot_release_x15}/
	wget --directory-prefix="/opt/source/u-boot_${u_boot_release_x15}/" ${rcn_https}/${u_boot_release_x15}/0001-beagle_x15-uEnv.txt-bootz-n-fixes.patch

	echo "u-boot_${u_boot_release} : /opt/source/u-boot_${u_boot_release}" >> /opt/source/list.txt
	echo "u-boot_${u_boot_release_x15} : /opt/source/u-boot_${u_boot_release_x15}" >> /opt/source/list.txt

	chown -R ${rfs_username}:${rfs_username} /opt/source/
}
```

If uboot gets compiled during the image creation from this source we could simply inject the patch at this point.  I don't think that is what is happening.

Still have to find where uboot gets added to the image so that we can have it pull from our compiled version instead.  That or run the recompile command on the source as listed above.

@codewithpassion 

- [x] Create a custom built process for u-boot that uses RCN's boot-builder scripts and applies the patch
- [x] Publish the custom u-boot .img and the registration file to a spot on S3
- [x] Add a tools/hwpack/openrov-beaglebone.conf based on the beaglebone.conf to omap-image-builder that specified an alternative conf_*= settings to point to our custom u-boot on S3
- [x] Build a new image and verify the new u-boot is active and that the image builds correctly.

**Comment by BrianAdams** [comment]
Okay, not sure how to validate the bootloader is the customized one without actively monitoring the serial  input for the message about the boot pausing for 5 seconds...

Did see this via dmesg:
```
[    4.163882] usb 1-1: Manufacturer: KYE Systems Corp.
[    4.163892] usb 1-1: SerialNumber: 200901010001
[    9.868018] EXT4-fs (mmcblk0p1): mounted filesystem with ordered data mode. Opts: (null)
```

**Comment by BrianAdams** [comment]
Verified patched in 30.0.3
